addons/mrp/models/mrp_production.py

```diff
@@ -1268,7 +1268,11 @@ def _set_qty_producing(self, pick_manual_consumption_moves=True):
         # waiting for a preproduction move before assignement
         is_waiting = self.warehouse_id.manufacture_steps != 'mrp_one_step' and self.picking_ids.filtered(lambda p: p.picking_type_id == self.warehouse_id.pbm_type_id and p.state not in ('done', 'cancel'))
 
-        for move in (self.move_raw_ids.filtered(lambda m: not is_waiting or m.product_id.tracking == 'none') | self.move_finished_ids.filtered(lambda m: m.product_id != self.product_id)):
+        for move in (
+            self.move_raw_ids.filtered(lambda m: not is_waiting or m.product_id.tracking == 'none')
+            | self.move_finished_ids.filtered(lambda m: m.product_id != self.product_id)
+            | self.move_finished_ids.filtered(lambda m: m.product_id == self.product_id and m.product_id.tracking == 'serial')
```

Comment:
Can't you join these two together?
Usually we try to avoid `filtered()` in a loop definition, because this does additional iterations to know on which records to iterate *again*. Although I guess it's a bit trickier since we join `move_raw_ids` and `move_finished_ids` in this case :slightly_frowning_face: 
But it would at least avoid looping twice through the `move_finished_ids` when we could do it only once.